src/main_solver.jl

```diff
@@ -0,0 +1,121 @@
+function solve!(params::ModelParams)
+    nb = params.nb # Number of blocks
+    x = params.x
+    y = params.y
+    y_L = params.y_L
+    y_U = params.y_U
+
+    ρ = deepcopy(params.opt.step_size_min) # initialize ρ as the minimum allowed step size value
+    μ = 10 # from Boyd's book (make options later)
+    τ = 2
+
+    iter_count = 0
+    params.opt.update_scheme == :JACOBI && (temp_x = zeros(Float64, length(x)))
+    elapsed_time = 0.0
+
+    while !(params.converged || params.tired)
+        iter_count += 1
+        params.opt.update_scheme == :JACOBI && (fill!(temp_x, 0.0))
+        max_iter_time = 0.0
+        params.init_obj_value = 0.0 # reset to zero
+
+        y_slice = @view y[params.model.problem_size.con_counter+1:end]
+
+        for i = 1:nb
+            params.method != DualDecomposition && update_primal!(params.init_blocks[i], x)
```

Comment:
I would suggest using a `if` statement there to improve readibility, instead of inlining everything